__tests__/integration/hathorwallet_facade.test.js

```diff
@@ -0,0 +1,1665 @@
+    expect(meltAuthorities).toHaveLength(1);
+
+    // Destroying all melt authorities
+    await waitUntilNextTimestamp(hWallet, destroyMeltTx);
+    const { hash: destroyAllMintTx } = await hWallet.destroyAuthority(tokenUid, 'melt', 1);
+    await waitForTxReceived(hWallet, destroyAllMintTx);
+    meltAuthorities = await hWallet.getMeltAuthority(tokenUid, { many: true });
+    expect(meltAuthorities).toHaveLength(0);
+
+    // Trying to melt and validating its error object
+    const meltFailure = await hWallet.meltTokens(tokenUid, 100)
+      .catch(err => err);
+    expect(meltFailure).toHaveProperty('success', false);
+    expect(meltFailure.message).toContain('authority output');
+  });
+});
+
+describe('createNFT', () => {
+  afterEach(async () => {
+    await stopAllWallets();
+    await GenesisWalletHelper.clearListeners();
+  });
+
+  it('should create an NFT with mint/melt authorities', async () => {
+    const hWallet = await generateWalletHelper();
+    await GenesisWalletHelper.injectFunds(hWallet.getAddressAtIndex(0), 100);
+
+    // Creating one NFT with default authorities
+    const nftTx = await hWallet.createNFT(
+      'New NFT',
+      'NNFT',
+      1,
+      sampleNftAddress,
+      {
+        createMint: true,
+        createMelt: true
+      },
+    );
+    expect(nftTx.hash).toBeDefined();
+    expect(nftTx.name).toEqual('New NFT');
+    expect(nftTx.symbol).toEqual('NNFT');
+    await waitForTxReceived(hWallet, nftTx.hash);
+
+    // Validating HTR fee payment
+    const htrBalance = await hWallet.getBalance(HATHOR_TOKEN_CONFIG.uid);
+    expect(htrBalance[0].balance.unlocked).toEqual(98); // 1 deposit, 1 fee
+    let nftBalance = await hWallet.getBalance(nftTx.hash);
+    expect(nftBalance[0].balance.unlocked).toEqual(1);
+
+    /*
+     * Since NFT authority tokens cannot be read by getMintAuthority, we will try to actually
+     * mint tokens to confirm the authority exists.
+     */
+    await waitUntilNextTimestamp(hWallet, nftTx.hash);
+    const rawMintTx = await hWallet.mintTokens(
+      nftTx.hash,
+      10,
+    );
+    expect(rawMintTx).toHaveProperty('hash');
+    await waitForTxReceived(hWallet, rawMintTx.hash);
+    nftBalance = await hWallet.getBalance(nftTx.hash);
+    expect(nftBalance[0].balance.unlocked).toEqual(11);
+
+    /*
+     * Since NFT authority tokens cannot be read by getMeltAuthority, we will try to actually
+     * melt tokens to confirm the authority exists.
+     */
+    await waitUntilNextTimestamp(hWallet, rawMintTx.hash);
+    const htrMelt = await hWallet.meltTokens(
+      nftTx.hash,
+      5,
+    );
+    expect(htrMelt).toHaveProperty('hash');
+    await waitForTxReceived(hWallet, htrMelt.hash);
+    nftBalance = await hWallet.getBalance(nftTx.hash);
+    expect(nftBalance[0].balance.unlocked).toEqual(6);
+  });
+
+  // TODO: Test NFTs without mint/melt authorities. Maybe use getUtxos when their tests are done.
```

Comment:
Yes, this is important. Also selecting the addresses

Reply:
Done on [0b15620d](https://github.com/HathorNetwork/hathor-explorer-service/pull/174/commits/0b15620dc769be4328b551a320ae3186dcb3a86c).